tests/unit/commands/local/lib/test_cfn_api_provider.py

```diff
@@ -1186,6 +1186,86 @@ def test_empty_integration_array(self):
         provider = ApiProvider(make_mock_stacks_from_template(template))
         self.assertIsNone(provider.api.cors)
 
+class TestExtractResourcesWithDisableAuthorizerFlag(TestCase):
+    @patch("samcli.lib.providers.cfn_api_provider.CfnApiProvider._get_route_function_name")
+    def test_extract_resources_apigw_v2_route(self, mock_get_route_function_name: Mock):
+        template = {
+            "Resources": {
+                "MyRoute": {
+                    "Type": "AWS::ApiGatewayV2::Route",
+                    "Properties": {
+                        "ApiId": {
+                            "Ref": "APIGatewayV2"
+                        },
+                        "AuthorizationType": "CUSTOM",
+                        "RouteKey": "Get /Test",
+                        "Target": "DummyTarget"
+                    }
+                },
+            }
+        }
+        mock_get_route_function_name.return_value = ("Dummy", "Values")
+
+        provider = CfnApiProvider()
+        mock_collector = Mock()
+        mock_collector.add_routes = Mock()
+        provider._extract_cfn_gateway_v2_route(stack_path="stack_path", resources={}, logical_id="MyRoute", route_resource=template["Resources"]["MyRoute"], collector=mock_collector, disable_authorizer=True)
+        assert mock_collector.add_routes.call_args.args[1][0].authorizer_name is None
```

Comment:
nit: can we change the asserts to use `unittest`'s assertions? The method call to make here would be `<mocked object>.assert_called_with` (ref: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with